source/common/grpc/async_client_impl.cc

```diff
@@ -166,6 +166,10 @@ void AsyncStreamImpl::streamError(Status::GrpcStatus grpc_status, const std::str
   resetStream();
 }
 
+void AsyncStreamImpl::onClosure() {
+  // No-op since stream closure is handled within other callbacks.
```

Comment:
Note that the way closure is previously/currently handled in this implementation is a little bit error-prone, since it's technically possible for the remote to close the stream prior to the local completing sending its request (unless grpc explicitly disallows this, which is beyond my familiarity with it).

Reply:
AFAIK it does disallow this.

Reply:
@goaway which implementation are you referring to? gRPC disallows that the client keeps stream open after server finishes the stream. Usually server will close a stream with a RST_STREAM in that case.

Reply:
@lizan that's what I wanted to confirm. Thanks! :)